### PR TITLE
Add test assertion diff examples

### PR DESCRIPTION
**Pull request by Dilhasha**
## Purpose
Add test assertion diff examples

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 1]
```diff
@@ -282,6 +282,153 @@ function foo() {
 }
 ```
 
+### **Difference between expected and actual values when using 'assertEquals'**
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:285]:
```suggestion
### **Difference between expected and actual values when using 'assertEquals'**
```
Titles should not be in bold.

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 2]
```diff
@@ -282,6 +282,153 @@ function foo() {
...
+
+* When type of the compared values are different,
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:287]:
```suggestion
* When type of the compared values are different.
```

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 3]
```diff
@@ -282,6 +282,153 @@ function foo() {
...
+
+@test:Config {}
+function testAssertStringAndInt() {
+    test:assertEquals(1, "1");
+}
+```
+
+***Output:***
+
+```
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:302]:
```suggestion
```bash
```

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 4]
```diff
@@ -282,6 +282,153 @@ function foo() {
...
+[fail] testAssertStringAndInt:
+    Assertion Failed!
+
+        expected: <string> '1'
+        actual  : <int> '1'
+```
+
+* For string typed values,
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:310]:
```suggestion
* For string-typed values.
```

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 5]
```diff
@@ -282,6 +282,153 @@ function foo() {
...
+
+@test:Config {}
+function testAssertString() {
+    test:assertEquals("hello ballerina user\nWelcome to Ballerina", "hello user\nWelcome to Ballerina");
+}
+```
+
+***Output:***
+
+```
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:325]:
```suggestion
```bash
```

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 6]
```diff
@@ -282,6 +282,153 @@ function foo() {
...
+
+         @@ -1,2 +1,2 @@
+
+         -hello user
+         +hello ballerina user
+         Welcome to Ballerina
+
+```
+
+* For json/record/map typed values,
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:347]:
```suggestion
* For JSON/record/map typed values.
```

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 7]
```diff
@@ -282,6 +282,153 @@ function foo() {
...
+            status: {pass: false}
+        }
+    };
+    test:assertEquals(j1, j2);
+}
+```
+
+***Output:***
+
+```
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:380]:
```suggestion
```bash
```

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 8]
```diff
@@ -282,6 +282,153 @@ function foo() {
...
+        expected value  : 35
+        actual value    : 99
+
+        key: marks.status.pass
+        expected value  : false
+        actual value    : true
+
+```
+
+* For other anydata typed values,
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:406]:
```suggestion
* For other anydata-typed values.
```

File: swan-lake/learn/testing-ballerina-code/writing-tests.md [hunk 9]
```diff
@@ -282,6 +282,153 @@ function foo() {
...
+function testAssertTuples() {
+    [int, string] a = [10, "John"];
+    [int, string] b = [12, "John"];
+    test:assertEquals(a, b);
+}
+```
+
+***Output:***
+
+```
```
Comment (praneesha) [inline swan-lake/learn/testing-ballerina-code/writing-tests.md:423]:
```suggestion
```bash
```

File: swan-lake/learn/testing-ballerina-code/writing-tests.md (hunk 1 above)
Comment (Dilhasha) [reply swan-lake/learn/testing-ballerina-code/writing-tests.md:285]: Ok noted, we might have to change all the other titles in this page too. Will do.

**Review by praneesha — APPROVED** [review]

**Pull request merged**